src/libraries/System.Collections/src/System/Collections/Generic/SortedDictionary.cs

```diff
@@ -947,6 +947,21 @@ public override int Compare(KeyValuePair<TKey, TValue> x, KeyValuePair<TKey, TVa
             {
                 return keyComparer.Compare(x.Key, y.Key);
             }
+
+            public override bool Equals(object? obj)
+            {
+                if (obj is KeyValuePairComparer other)
+                {
+                    // Commonly, both comparers will be the default comparer (and reference-equal). Avoid a virtual method call to Equals() in that case.
```

Comment:
Since this method is only called once by the constructor, I don't think avoiding a virtual call is too important here. Unless we have precedent in other parts of System.Collections, I would say keep things simple.

Reply:
This comment and approach is copied from the following, it might make sense to just do the reference equality check, let me know what you'd prefer. https://github.com/dotnet/runtime/blob/main/src/libraries/System.Collections/src/System/Collections/Generic/SortedSet.cs#L853

Reply:
In that case we could keep it for the sake of consistency I suppose.